src/Symfony/Component/Yaml/Tag/TagResolver.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Yaml\Tag;
+
+use Symfony\Component\Yaml\Yaml;
+use Symfony\Component\Yaml\Exception\UnsupportedTagException;
+
+/**
+ * @author Guilhem N. <egetick@gmail.com>
+ *
+ * @internal
+ */
+final class TagResolver
```

Comment:
looking at this closer, we might not need it: we could simply feed `Yaml\Parser` with an associative array of tags-to-tagresolvers:
`new Parser(..., ..., array('iterator' => new IteratorTagResolver())` (prefix excluded, or auto added)

we don't need to handle the "priority" + "supports()" logic, it just adds complixity to me, no flexibility